src/main/java/core/basesyntax/service/impl/TransactionProcessorImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.Operation;
+import core.basesyntax.dto.ProductTransaction;
+import core.basesyntax.service.TransactionProcessor;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.handler.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class TransactionProcessorImpl implements TransactionProcessor {
+    private final Map<Operation, OperationHandler> operationHandlers;
+
+    public TransactionProcessorImpl(
+            Map<Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void process(List<ProductTransaction> productTransactions) {
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+        OperationHandler operationHandler;
```

Comment:
Why new Strategy object every time? make it a field or just remove that class at all and use the map directly, because basically all it does is encapsulates this map and I do not really think it is necessary